src/ClientBuilder.php

```diff
@@ -13,10 +16,12 @@ public static function build(
         array $tokenOptions,
         Cache $cache,
         array $guzzleOptions = null
-    ): \GuzzleHttp\Client {
+    ): Client {
         $cacheHandler = new AccessTokenCacheHandler($cache);
 
-        $stack = static::getStack();
+        $stack = $guzzleOptions['handler'] ?? HandlerStack::create();
```

Comment:
I think it would be best to execute the logic to create `HandlerStack` only if `$options['handler']` is not defined. If it is, it will be automatically merged in the client.

Reply:
After talking with @akira28, I suggest the following:
* if we pass `$options['handler']`, the handler should be overridden
* We should pass another parameter (something like `middlewares`), we should add middlewars.

What do you think @akira28 @joskfg @xaviapa ?

Reply:
I don't like it because in guzzle doesn't exist middleware and a handler stack. The handler stack is a stack of middlewares executed to resolve the the request, so they are the same.

If you customize the stack, you are specifying manually all the stack in the guzzle. Why would you pass a setting called middleware? Should it be used a whole stack? Should you add it after all the default stack? Or before?

Reply:
I agree with @joskfg. My intention here was to just allow people passing their own stack (with related middlewares) instead of creating a new one each time

Reply:
I do not fully understand your comment @joskfg : one one hand this PR is already merging current handler with injected one (middlewares will be added *after* the ones that have been passed) and you've accepted the changes, on the other hand you are saying you dislike such behavior.

Let's remember this method is just a builder to create a client with such oauth2 middleware: I would recommend building it manually if the needed middleware is a specific use case, but we can add another builder (or a flag to the current one, but it seems you dislike this) if the use case is generic enough.

What is the intended use case @akira28 ?

Reply:
the intended use case is to have the ability to add another middleware. IMHO this is the simplest way to achieve what I need, maintaining retrocompatibility, and adding the ability to have a 'custom' stack containing my own middlewares

Reply:
Why has this been merged :unamused: ? We still had a discussion ongoing, now we have potentially released a breaking change if we want to take another path

Reply:
the current version is not breaking. As discussed, we can create a breaking 2.0 changing (and improving) this helper

Reply:
That's right, I did not explain myself correctly. We haven't actually released a breaking change, but hastily releasing this change would cause us to release a BC change in the near future if the approach we want to follow is another.